### PR TITLE
Added Helm Chart Release CI

### PR DESCRIPTION
**Pull request by opdude**
It seems there is no current release pipeline for the NSQ helm chart. This PR adds some CI based on what [beeinventor](https://github.com/BeeInventor/charts) had previously.

This PR requires that a gh-pages branch is created on the repo (https://jiafulow.github.io/blog/2020/07/09/create-gh-pages-branch-in-existing-repo/)

- Lint checks on PRs
- Releases on pushes to the master branch
- Moved chart into a charts folder

### CODE REVIEW

**Comment by ploxiln** [comment]
any opinions @LordNoteworthy ? care to review?

File: .github/workflows/lint-helm.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  pull_request_review:
+    types: [submitted, edited]
+  release:
+    types:
+    - published
+    - created
+    - edited
+jobs:
+  lint-helm:
+    runs-on: ubuntu-latest
```
Comment (ploxiln) [inline .github/workflows/lint-helm.yaml]: Mind setting this to ubuntu-20.04 for now, and we can update explicitly later when a new version is available?

**Comment by ayoubfaouzi** [comment]
LGTM

File: .github/workflows/lint-helm.yaml (hunk 1 above)
Comment (opdude) [reply .github/workflows/lint-helm.yaml]: Done :)

**Review by ayoubfaouzi — APPROVED** [review]

**Comment by ploxiln** [comment]
Empty gh-pages branch created.

**Comment by ploxiln** [comment]
looks like it worked, https://nsqio.github.io/helm-chart/index.yaml is live

**Pull request merged**